witchcraft/server_metrics.go

```diff
@@ -141,14 +141,31 @@ func (s *Server) initMetrics(ctx context.Context, installCfg config.Install) (rR
 	}, nil
 }
 
-func (s *Server) filterMetricValues(metricID string, metricVal metrics.MetricVal) map[string]interface{} {
+func (s *Server) collectMetricValues(metricName string, metricVal metrics.MetricVal) map[string]interface{} {
+	valuesToUse := make(map[string]interface{})
+	s.visitMetricValues(metricName, metricVal, func(key string) {
+		valuesToUse[key] = metricVal.Value(key)
+	})
+	return valuesToUse
+}
+
+// visitMetricValues calls the provided visit function for each key in the provided metric value,
+// If the metric name or key is blacklisted, the visit function is not called.
```

Comment:
Documentation could be a bit clearer

```suggestion
// visitMetricValues calls the provided visit function for each key in the provided metric value.
// If the metric with the provided metricID is in the metrics blacklist, the visit function is not called for any keys.
// The visit function is not called for keys that are in the metric type value blacklist for the type.
```